File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(arr) {


    let n = arr.length
    let i = 0
    while( i < n) {
        let correctIndex = arr[i]

        if (arr[i]<arr.length && arr[i] != arr[correctIndex]) {
            let temp = arr[i]
            arr[i] = arr[correctIndex]
            arr[correctIndex] = temp
            correctIndex = arr[i]
        }else{
            i++
        }

    }
    console.log(arr)
    let j = 0
    for(j ; j<arr.length ; j++){
        if(arr[j] != j){
            return j
        }
    }

    return j




    // let set = new Set(nums)

    // for(let i = 0 ; i<= nums.length ; i++){
    //     if(!set.has(i)){
    //         return i
    //     }
    // }
    // return 1

};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int sum = 0;

        for(int i = 0 ; i<nums.length; i++){
            sum += nums[i];
        }   


        int mul = nums.length * (nums.length+1);

        System.out.println(nums.length);
        System.out.println(mul);
        int hulf = mul/2;
        System.out.print(hulf);
        return hulf-sum;
    }
}